src/diffusion/model.py

```diff
@@ -134,7 +136,10 @@ def __init__(
     def configure_optimizers(self):
         # Fully control LR from scheduler
         optimizer = torch.optim.AdamW(self.parameters(), lr=1.0, weight_decay=self.weight_decay)
-        scheduler_lambda = partial(rsqrt_with_warmup, max_lr=self.lr, min_lr=self.min_lr, warmup=self.warmup)
+        if self.decay_range is not None:
```

Comment:
Well, that is not good criteria for choosing a scheduler. I suggest small refactoring:
1. Let's replace all optimizer-related arguments in `__init__` with `optimizer_kwargs`. And save this dict as it is (`self.opt_kwargs = opt_kwargs`)
2. In `configure_optimizer` extract necessary values from dict (e.g. `max_lr=self.opt_kwargs["max_lr"])
3. Use`scheduler` key to extract the name of the scheduler
4. In config files move all optimizer-related arguments into a separate block
5. In main script pass parsed block directly to DiDi, i.e. `DiDi(..., opt_kwargs=config.optimizer, **config.didi)`